drools-core/src/main/java/org/drools/core/command/runtime/pmml/ApplyPmmlModelCommand.java

```diff
@@ -119,6 +127,36 @@ public void setOutIdentifier(String outIdentifier) {
     
     @Override
     public PMML4Result execute(Context context) {
+        if (requestData == null) {
+            throw new IllegalStateException("ApplyPmmlModelCommand requires request data (PMMLRequestData) to execute");
+        }
+        final PMMLConstants toInvoke = getToInvoke(context);
+        switch (toInvoke) {
+            case NEW:
+                return executePMMLTrusty();
+            case LEGACY:
+                return executePMMLLegacy(context);
+            default:
+                throw new RuntimeException("Unmanaged PMMLConstants " + toInvoke);
+        }
+    }
+
+    protected PMMLConstants getToInvoke(Context context) {
+        ClassLoader classLoader = getClassLoader((RegistryContext)context);
+        return toEnable(classLoader);
+    }
+
+    private ClassLoader getClassLoader(RegistryContext registryContext) {
+        KieSession kieSession = registryContext.lookup(KieSession.class);
+        InternalKnowledgeBase kieBase = (InternalKnowledgeBase) kieSession.getKieBase();
+        return kieBase.getRootClassLoader();
```

Comment:
What about something similar to
```suggestion
        try {
            KieSession kieSession = registryContext.lookup(KieSession.class);
            InternalKnowledgeBase kieBase = (InternalKnowledgeBase) kieSession.getKieBase();
            return kieBase.getRootClassLoader();
        catch (Exception e) {
            logger.warn("Impossible to retrieve RootClassLoader, using ContextClassLoader [" + e.getMessage() + "]", e);
            return Thread.currentThread().getContextClassLoader();
        }
```

Reply:
@danielezonca 
Done